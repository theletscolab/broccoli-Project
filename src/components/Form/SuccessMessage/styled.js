import styled, { css } from 'styled-components/macro'

const StyledSuccessMessage = styled.form`
  padding: 3.5rem 2.5rem;
  width: 300px;
  ${({ theme }) => css`
    ${theme.mediaQueries.md} {
      width: 375px;
    }
  `};
`
const MessageTitle = styled.h2`
  display: block;
  color: ${({ theme }) => theme.colors.grey900};
  font-size: ${({ theme }) => theme.fontSizes[2]};
  font-weight: ${({ theme }) => theme.fontWeights.bold};
  line-height: 1.31;
  margin-bottom: 3rem;
  position: relative;

  &::after {
    background-color: ${({ theme }) => theme.colors.grey400};
    bottom: -1rem;
    content: ' ';
    height: 0.2rem;
    left: calc(50% - 2.5rem);
    position: absolute;
    width: 50px;
  }
`

const Message = styled.div`
  color: ${({ theme }) => theme.colors.grey700};
  padding: 0.6rem;
  text-align: center;
  margin-bottom: 3rem;
`

export { StyledSuccessMessage, MessageTitle, Message }
