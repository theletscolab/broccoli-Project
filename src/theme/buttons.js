export const buttonSizes = {
  small: {
    height: '42px',
    fontSize: '1.4rem',
    padding: '1.2rem 1.7rem'
  },
  medium: {
    height: '48px',
    fontSize: '1.6rem',
    padding: '1.4rem 2.4rem'
  },
  large: {
    height: '54px',
    fontSize: '1.8rem',
    padding: '1.6rem 2.8rem'
  }
}
