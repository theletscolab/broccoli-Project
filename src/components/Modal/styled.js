import styled from 'styled-components/macro'

const ModalOverlay = styled.div`
  align-items: center;
  bottom: 0;
  background-color: ${({ theme }) => theme.colors.black50};
  display: flex;
  justify-content: center;
  left: 0;
  overflow-y: auto;
  position: fixed;
  right: 0;
  top: 0;
  height: ${({ modal }) => (modal ? 'auto' : 0)};
  width: ${({ modal }) => (modal ? 'auto' : 0)};
  opacity: ${({ modal }) => (modal ? 1 : 0)};
  transition: all 0.3s;
`

const ModalStyled = styled.div`
  background-color: ${({ theme }) => theme.colors.white};
  border: 2px solid ${({ theme }) => theme.colors.black};
  position: relative;
  z-index: 100;
`

const ModalCloseBtn = styled.button`
  background: none;
  border: 0;
  color: ${({ theme }) => theme.colors.grey700};
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 2.4rem;
  line-height: 1;
  height: 32px;
  position: absolute;
  right: 0;
  top: 0;
  width: 32px;

  &:hover {
    background-color: ${({ theme }) => theme.colors.grey200};
  }

  &:focus {
    outline: 2px dotted ${({ theme }) => theme.colors.grey700};
    outline-offset: 2px;
  }
`

export { ModalOverlay, ModalStyled, ModalCloseBtn }
