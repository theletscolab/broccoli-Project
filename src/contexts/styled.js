import styled from 'styled-components/macro'

export const ModalContextStyled = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  justify-content: center;
  margin-top: 62px;
  text-align: center;
`