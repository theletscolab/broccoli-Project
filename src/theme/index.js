import { buttonSizes } from './buttons'
// App theme

// Colors Palette
const awlxDarkBlue = '#612fff'
const awlxLightBlue = '#e9e3ff'
const white = '#ffffff'
const grey50 = '#fafafa'
const grey100 = '#f5f5f5'
const grey200 = '#eeeeee'
const grey300 = '#e0e0e0'
const grey400 = '#bdbdbd'
const grey500 = '#9e9e9e'
const grey600 = '#757575'
const grey700 = '#616161'
const grey800 = '#424242'
const grey900 = '#212121'
const black = '#000000'
const black15 = 'rgba(0, 0, 0, 0.15)'
const black25 = 'rgba(0, 0, 0, 0.25)'
const black50 = 'rgba(0, 0, 0, 0.50)'
const black75 = 'rgba(0, 0, 0, 0.75)'
const error = '#cc0000'
const bodyBg = white
const bodyColor = grey900

// Responsive breakpoints and media-queries (mobile first).
const breakpoints = ['576px', '768px', '992px', '1200px']

const fonts = {
  base: `'Montserrat', sans-serif`,
  caveat: `'Caveat', cursive`
}

const fontWeights = {
  light: 300,
  regular: 400,
  medium: 500,
  semiBold: 600,
  bold: 700
}

const headingFontFamily = fonts.base

const fontSizes = ['1.2rem', '1.6rem', '2rem', '2.4rem', '3.2rem', '4.8rem', '6.4rem']

const bodyText = fontSizes[1]
const headingWeight = fontWeights.bold
const headingLineHeight = 1.4

const theme = {
  breakpoints,
  mediaQueries: {
    sm: `@media screen and (min-width: ${breakpoints[0]})`,
    md: `@media screen and (min-width: ${breakpoints[1]})`,
    lg: `@media screen and (min-width: ${breakpoints[2]})`,
    xl: `@media screen and (min-width: ${breakpoints[3]})`
  },
  colors: {
    awlxDarkBlue,
    awlxLightBlue,
    white,
    grey50,
    grey100,
    grey200,
    grey300,
    grey400,
    grey500,
    grey600,
    grey700,
    grey800,
    grey900,
    black,
    black15,
    black25,
    black50,
    black75,
    bodyBg,
    bodyColor,
    error
  },
  space: [0, 4, 8, 16, 32, 64, 128, 256],
  fonts,
  fontSizes,
  bodyText,
  fontWeights,
  headingWeight,
  headingLineHeight,
  headingFontFamily,
  shadows: {
    small: '0 0 4px rgba(0, 0, 0, .125)',
    large: '0 0 24px rgba(0, 0, 0, .125)'
  },
  maxContainerWidth: '992px',
  buttonSizes
}

export default theme
