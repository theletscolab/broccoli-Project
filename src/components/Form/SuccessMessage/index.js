import React from 'react'
import PropTypes from 'prop-types'
import { ModalContext } from '../../../contexts/modalContext'
import Button from '../../Button'
import { StyledSuccessMessage, MessageTitle, Message } from './styled'

const SuccessMessage = ({ formTitle, formMessage, formBtnTxt }) => {
  let { handleModal } = React.useContext(ModalContext)
  return (
    <StyledSuccessMessage>
      <MessageTitle>{formTitle}</MessageTitle>
      <Message>{formMessage}</Message>
      <Button title={formBtnTxt} onClick={() => handleModal()} />
    </StyledSuccessMessage>
  )
}

SuccessMessage.propTypes = {
  formTitle: PropTypes.string.isRequired,
  formMessage: PropTypes.string.isRequired,
  formBtnTxt: PropTypes.string.isRequired
}

SuccessMessage.defaultProps = {
  formTitle: 'All done!',
  formMessage: 'You will be one of the first to experience Broccoli & Co. when we launch.',
  formBtnTxt: 'OK'
}

export default SuccessMessage
