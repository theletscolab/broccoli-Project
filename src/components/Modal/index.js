import React from 'react'
import { ModalContext } from '../../contexts/modalContext'
import { ModalOverlay, ModalStyled, ModalCloseBtn } from './styled'

const Modal = () => {
  let { modalContent, handleModal, modal } = React.useContext(ModalContext)
  return (
    modal && (
      <ModalOverlay modal={modal}>
        <ModalStyled>
          <ModalCloseBtn onClick={() => handleModal()}>x</ModalCloseBtn>
          {modalContent}
        </ModalStyled>
      </ModalOverlay>
    )
  )
}

export default Modal
