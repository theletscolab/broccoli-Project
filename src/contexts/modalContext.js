import React from 'react'
import { useModal } from '../hooks/useModal'
import Modal from '../components/Modal'
import { ModalContextStyled } from './styled'

let ModalContext
let { Provider } = (ModalContext = React.createContext())

let ModalProvider = ({ children }) => {
  let { modal, handleModal, openModal, closeModal, modalContent } = useModal()
  return (
    <ModalContextStyled>
      <Provider value={{ modal, handleModal, openModal, closeModal, modalContent }}>
        <Modal />
        {children}
      </Provider>
    </ModalContextStyled>
  )
}

export { ModalContext, ModalProvider }
