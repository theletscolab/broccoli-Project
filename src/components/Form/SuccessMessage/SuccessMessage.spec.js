import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../../../theme'
import SuccessMessage from './index'
import { ModalProvider } from '../../../contexts/modalContext'
import Modal from '../../../components/Modal'

describe('components > Form > SuccessMessage', () => {
  let component

  describe('with defaults', () => {
    beforeAll(() => {
      component = render(<SuccessMessage />, {
        wrapper: ({ children }) => {
          return (
            <ThemeProvider theme={theme}>
              <ModalProvider>
                <Modal />
                {children}{' '}
              </ModalProvider>
            </ThemeProvider>
          )
        }
      })
    })

    it('rendered without crashing', () => {
      expect(component.asFragment()).toMatchSnapshot()
    })
  })
})
