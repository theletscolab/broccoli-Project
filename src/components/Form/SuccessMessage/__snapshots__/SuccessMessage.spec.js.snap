// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Form > SuccessMessage with defaults rendered without crashing 1`] = `
<DocumentFragment>
  <div
    class="styled__ModalContextStyled-sc-6xjet7-0 kWmZTI"
  >
    <form
      class="styled__StyledSuccessMessage-sc-fz2q1d-0 fdnjXM"
    >
      <h2
        class="styled__MessageTitle-sc-fz2q1d-1 elqgCX"
      >
        All done!
      </h2>
      <div
        class="styled__Message-sc-fz2q1d-2 PDMrt"
      >
        You will be one of the first to experience Broccoli & Co. when we launch.
      </div>
      <button
        class="styled__StyledButton-sc-1yy0gyz-0 kSrkyH"
      >
        OK
      </button>
    </form>
     
  </div>
</DocumentFragment>
`;
